runtime/zeitgeist/src/xcm_config/config.rs

```diff
@@ -0,0 +1,305 @@
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+use super::fees::{native_per_second, FixedConversionRateProvider};
+use crate::{
+    AccountId, Ancestry, AssetManager, AssetRegistry, Balance, Call, CurrencyId, MaxInstructions,
+    Origin, ParachainInfo, ParachainSystem, PolkadotXcm, RelayChainOrigin, RelayNetwork,
+    UnitWeightCost, UnknownTokens, XcmpQueue, ZeitgeistTreasuryAccount,
+};
+
+use alloc::vec::Vec;
+use frame_support::{parameter_types, traits::Everything};
+use orml_asset_registry::{AssetRegistryTrader, FixedRateAssetRegistryTrader};
+use orml_traits::{location::AbsoluteReserveProvider, MultiCurrency};
+use orml_xcm_support::{
+    DepositToAlternative, IsNativeConcrete, MultiCurrencyAdapter, MultiNativeAsset,
+};
+use pallet_xcm::XcmPassthrough;
+use polkadot_parachain::primitives::Sibling;
+use sp_runtime::traits::Convert;
+use xcm::{
+    latest::{
+        prelude::{AccountId32, AssetId, Concrete, GeneralKey, MultiAsset, NetworkId, X1, X2},
+        Junction, MultiLocation,
+    },
+    opaque::latest::Fungibility::Fungible,
+};
+use xcm_builder::{
+    AccountId32Aliases, AllowKnownQueryResponses, AllowSubscriptionsFrom,
+    AllowTopLevelPaidExecutionFrom, FixedRateOfFungible, FixedWeightBounds, LocationInverter,
+    ParentIsPreset, RelayChainAsNative, SiblingParachainAsNative, SiblingParachainConvertsVia,
+    SignedAccountId32AsNative, SignedToAccountId32, SovereignSignedViaLocation, TakeRevenue,
+    TakeWeightCredit,
+};
+use xcm_executor::Config;
+use zeitgeist_primitives::types::Asset;
+
+pub mod zeitgeist {
+    #[cfg(test)]
+    pub const ID: u32 = 2101;
+    pub const KEY: &[u8] = &[0, 1];
+}
+
+pub struct XcmConfig;
+
+/// The main XCM config
+/// This is where we configure the core of our XCM integrations: how tokens are transferred,
+/// how fees are calculated, what barriers we impose on incoming XCM messages, etc.
+impl Config for XcmConfig {
+    /// The handler for when there is an instruction to claim assets.
+    type AssetClaims = PolkadotXcm;
+    /// How to withdraw and deposit an asset.
+    type AssetTransactor = MultiAssetTransactor;
+    /// The general asset trap - handler for when assets are left in the Holding Register at the
+    /// end of execution.
+    type AssetTrap = PolkadotXcm;
+    /// Additional filters that specify whether the XCM instruction should be executed at all.
+    type Barrier = Barrier;
+    /// The outer call dispatch type.
+    type Call = Call;
+    /// Combinations of (Location, Asset) pairs which are trusted as reserves.
+    // Trust the parent chain, sibling parachains and children chains of this chain.
+    type IsReserve = MultiNativeAsset<AbsoluteReserveProvider>;
+    /// Combinations of (Location, Asset) pairs which we trust as teleporters.
+    type IsTeleporter = ();
+    /// Means of inverting a location.
+    type LocationInverter = LocationInverter<Ancestry>;
+    // How to get a call origin from a `OriginKind` value.
```

Comment:
Some of the comments are `//` , better we keep all of them `///`

Reply:
Yeah the comment for `OriginConverter` should be a docstring actually. Fixed in https://github.com/zeitgeistpm/zeitgeist/pull/661/commits/fc6ae1930f7c350a3ec8d7b0edf1ab6e63b195c8
However, the other comments are intended as comments because they actually describe an implementation detail in contrast to a documentation about the purpose of the type itself.